### PR TITLE
Saxon done ✓

### PR DESCRIPTION
**Pull request by TaliaMalia**

### CODE REVIEW

File: babel.config.js
Comment (musoke) [inline babel.config.js:1]: Was this needed to get tests to run?  I don't think it should be necessary.

File: jest.config.cjs
Comment (musoke) [inline jest.config.cjs:1]: This also seems like it should be unnecessary

File: package.json
Comment (musoke) [inline package.json:1]: I don't think any of these changes are needed

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ Javascript utilities for working with rock climbing grades.
 - [x] French
 - [x] UIAA
 - [x] Ewbanks
+- [x] Saxon
```
Comment (musoke) [inline README.md:16]: Nice, good to remember this

File: src/__tests__/scales/saxon.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      const highGrade = Saxon.getScore('3')
+      const lowGrade = Saxon.getScore('2')
+      expect(highGrade[0] < lowGrade[1] && highGrade[0] > lowGrade[0])
+      if (lowGrade[1] !== undefined && highGrade[1] !== undefined) {
+        expect(highGrade[1]).toBeGreaterThan(lowGrade[1])
+      }
+    })
+  })
+
+  describe('Invalid grade format', () => {
```
Comment (musoke) [inline src/__tests__/scales/saxon.ts:35]: @TaliaMalia , please add a check that a grade such as 5a is disallowed.

**Review by musoke — CHANGES_REQUESTED** [review]
@TaliaMalia, most of this looks good!  I have left some comments below.  In particular, the changes you made to package.json, yarn.lock, jest.config and babel.config.js aren't needed.  

Could you also add a test for an edge case grade like 5a?  It should be disallowed.

File: babel.config.js
Comment (TaliaMalia) [reply babel.config.js:1]: Yes, I thought it would solve the error that I got, but the babel and jest config are actually unnecessary. I still don't know why I got that error :/

Comment (musoke) [reply babel.config.js:1]: Ok!

Comment (musoke) [reply babel.config.js:1]: I also can't tell why you got that error.  Maybe you forgot to `yarn install`?

Comment (TaliaMalia) [reply babel.config.js:1]: That's too bad, but thanks anyway! No, I don't think that was the problem.

File: jest.config.cjs
Comment (TaliaMalia) [reply jest.config.cjs:1]: I deleted the babel config and changed the jest config to how it was before.

File: package.json
Comment (TaliaMalia) [reply package.json:1]: I undid my changes in the package.json.

File: src/__tests__/scales/saxon.ts (hunk 2 above)
Comment (TaliaMalia) [reply src/__tests__/scales/saxon.ts:35]: I added a test called "invalid character" that disallows to use grades < 7 to have letters from a to c.
While doing that I recognized that the saxonGradeRegex I wrote was incorrect and did not consider that the grades 1-6 are not allowed to have letters. I changed that now and everything works as it should.

File: src/__tests__/scales/saxon.ts [hunk 3]
```diff
@@ -70,6 +70,11 @@ describe('Saxon', () => {
       )
       expect(invalidGrade).toEqual(-1)
     })
+    test('invalid character', () => {
+      const invalidGrade = Saxon.getScore('5a')
+      expect(console.warn).toHaveBeenCalledWith('Unexpected grade format: 5a for grade scale Saxon')
```
Comment (musoke) [inline src/__tests__/scales/saxon.ts:75]: good

File: src/scales/saxon.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import routes from '../data/routes.json'
 import { Route } from '.'
 import { GradeBandTypes, routeScoreToBand } from '../GradeBands'
 
-const saxonGradeRegex = /^(([1-9]|1[0-3])([a-c])?){1}(?:(\/)(([1-9]|1[0-3])([a-c])?))?$/i
+const saxonGradeRegex = /^((([7-9]|1[0-3])([a-c]))|([1-6]))$/i
```
Comment (musoke) [inline src/scales/saxon.ts:6]: nice

File: src/scales/ewbank.ts [hunk 5]
```diff
@@ -26,17 +26,18 @@ const EwbankScale: GradeScale = {
...
 
     if (typeof score === 'number') {
-      return routes[validateScore(score)].ewbank
+      return validateScore(score)
     }
 
-    const low: string = routes[validateScore(score[0])].ewbank
-    const high: string = routes[validateScore(score[1])].ewbank
+    const low: string = validateScore(score[0])
+    const high: string = validateScore(score[1])
```
Comment (musoke) [inline src/scales/ewbank.ts:40]: One last question: could you say again why this change was necessary?  The tests run for me without it.

File: src/scales/uiaa.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ const UIAAScale: GradeScale = {
   displayName: 'UIAA Scale',
   name: GradeScales.UIAA,
   offset: 2000,
-  allowableConversionType: [GradeScales.YDS],
+  allowableConversionType: [GradeScales.YDS, GradeScales.SAXON, GradeScales.EWBANK, GradeScales.FRENCH],
```
Comment (musoke) [inline src/scales/uiaa.ts:17]: Good catch, this was a bug we hadn't noticed.

**Review by musoke — CHANGES_REQUESTED** [review]

**Comment by musoke** [comment]
I left one more question for you.
I also pushed some changes to make ts-standard happy, so you'll have to pull before doing further edits (if any).

File: src/scales/ewbank.ts (hunk 5 above)
Comment (TaliaMalia) [reply src/scales/ewbank.ts:40]: Yes of course. If I don't use type assertion in validateScore I get a type error **"Type 'string | number' is not assignable to type 'string'. Type 'number' is not assignable to type 'string'"**. I really can't explain to you why that only happens in ewbank or why that happens at all...

**Comment by musoke** [comment]
I can't reproduce that error. Does it still happen if you undo the change?

**Comment by TaliaMalia** [comment]
Yes, the only way the code is working for me is when I run it like I committed it. If it does not have any consequences and the code is still working without my changes I can just undo the changes and push.

**Comment by musoke** [comment]
Ok, it sounds like the version you wrote is the only one that works for both of us.  I just wish I knew why so we could avoid such inconsistencies in future.

**Review by musoke — APPROVED** [review]

**Comment by musoke** [comment]
I'll squash and merge shortly

**Comment by TaliaMalia** [comment]
I wish I knew too. I checked everything I can think of, but just couldn't figure it out :/

**Comment by musoke** [comment]
Thank you for you contribution!

@all-contributors add @TaliaMalia  for code

**Comment by TaliaMalia** [comment]
Your welcome!
Thank you, that you were so patient and communicative :)
